### ISSUE TITLE
Inform user when a draft is already in progress

### ISSUE THREAD

**Issue by verythorough** [netlify/netlify-cms#1453]
**- Do you want to request a *feature* or report a *bug*?**
Feature

**- What is the current behavior?**
When editorial workflow is enabled, a user can open an existing file for editing without realizing another draft edit is already in progress. On the backend, both edits will be added to the same branch, and merged in the same PR. This can result in a user publishing more changes than expected.

**- What is the desired behavior?**
I could see a few improvements that could be useful here:
- In the collection list, add a tag to items currently in the editorial workflow, with their current status.
- When a user opens an existing file with an existing draft edit, show a message letting them know this.
- When the message appears to say a draft is already in progress, give the user the option to add their changes to that draft, or create a new one (with a new branch name) - potential for merge conflicts certainly goes up with this one, though.

**Comment by erquhart** [comment]
Totally agree. Maybe the third bullet would be more feasible when we have some sort of visual diffing for conflict resolution.

**Comment by rdebeasi** [comment]
I'm not sure if this is changed in 2.0 or if our editorial workflow configuration is different, but when I try to edit a file that someone else has already begun editing, I get this error message:

> Failed to persist entry: Error: Editorial workflow branch changed unexpectedly.

Anyway, thank you for logging this issue!

**Comment by erquhart** [comment]
@rdebeasi I can't reproduce that, can you open a new issue with steps to reproduce?

**Comment by rdebeasi** [comment]
Sure, will do!

**Comment by barthc** [comment]
@erquhart  can we close this?

**Comment by verythorough** [comment]
@barthc As far as I know, the problem described hasn't been solved, so I don't see why we'd close this unless it's being addressed in another issue (which case we should link it before we close this).

**Comment by barthc** [comment]
@verythorough 

> When editorial workflow is enabled, a user can open an existing file for editing without realizing another draft edit is already in progress.

The current behavior of the CMS in editorial mode when a user clicks on any entry for editing, the CMS first checks if there is a draft of the entry in editorial workflow and loads that instead. If not, it's creates a new draft from the existing file.

**Comment by verythorough** [comment]
@barthc 

> The current behavior of the CMS in editorial mode when a user clicks on any published entry for editing, the CMS first checks if there is a draft of the entry in editorial workflow and loads that instead. If not, it's creates a new draft from the existing file.

Correct! This has been true for as long as the editorial workflow has existed. The trouble is that the user doesn't know this is happening. So we have had situations where:

1. User A opens Doc Y for editing, adding a section for a new feature to be released in a week. They leave the status as "draft" because they plan to make more changes, and the change shouldn't be published until the feature is released, anyway.
2. User B opens Doc Y to fix a typo in a different section of the doc. User B has no idea that User A has changes in progress in Doc Y, and thinks that when they click "Publish" they will only be publishing their typo fix. Unintended publishing happens, and the world tragedy ensues!

**Comment by barthc** [comment]
@verythorough  Never thought the issue was referring to multiple users editing the same entry. Thanks for clarifying.

**Comment by verythorough** [comment]
Ah, got it! 👍
